app/main.py

```diff
@@ -2,0 +1,32 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+
+        self.alive.append(self)
```

Comment:
Appending `self` to `self.alive` should be `Animal.alive.append(self)` to ensure that the instance is added to the class-level list, not an instance-level list.